src/services/vault/index.ts

```diff
@@ -49,9 +50,11 @@ export const createVaultClient = memoize(async (vaultOptions: VaultOptions = {})
 
 export const loadVaultSecret =
 	(vaultClient: ReturnType<typeof createVaultClient> = createVaultClient()) =>
-	async (path: string) => {
+	async <T = Record<string, string>>(path: string): Promise<T> => {
```

Comment:
started something, didn't really finish it cognisant of scope, but should enable vault secret validation in the future